main/autogen/autogen.cc

```diff
@@ -224,6 +227,12 @@ class AutogenWalk {
             return;
         }
 
+        auto entry = make_pair(tree.loc(), original.symbol);
+        if (seenRefsByLoc.contains(entry)) {
+            return;
+        }
+        seenRefsByLoc.emplace(move(entry));
```

Comment:
WDYT about also skipping zero-length locs?  I guess that wouldn't really help with the problem being addressed here, because we probably keep the full-width locs around so that error messages from the duplicated constants are reasonable.

This would probably be epsilon faster if you used `insert`:

https://github.com/abseil/abseil-cpp/blob/2a18ba753bca64e79f6d9857a9f64e638b32c371/absl/container/flat_hash_set.h#L250-L261

so that you can do the lookup and (possible) insertion in a single go.

Reply:
> WDYT about also skipping zero-length locs? I guess that wouldn't really help with the problem being addressed here, because we probably keep the full-width locs around so that error messages from the duplicated constants are reasonable.

going to punt on this because I think we want this in, and I think that the current solution is "good enough." Happy to revisit if the difference starts to matter in practice.